### PR TITLE
ENG-722 usage of native widget configs in widget form

### PR DESCRIPTION
**Pull request by jeffgo10**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **jeffgo10**

If they are not already assigned, you can assign the PR to them by writing `/assign @jeffgo10` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@jeffgo10: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | bc2285a8bcedb06ea0e745b887b57d462ec49310 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/app-builder/PR-838/2.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/helpers/conversion.js
Comment (gidesan) [inline src/helpers/conversion.js:11]: please follow the eslint guideline

File: src/helpers/conversion.js
Comment (gidesan) [inline src/helpers/conversion.js:1]: let's use `convertJSONtoWidgetConfigNotation` so we know what it is

**Review by gidesan — CHANGES_REQUESTED** [review]

File: src/helpers/conversion.js
Comment (jeffgo10) [reply src/helpers/conversion.js:11]: @gidesan it's hard for me to follow that rule because `convertObject` and `convertArray` are both recursive functions that actually invokes each other.

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/838#pullrequestreview-488648082" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
